<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Givanna Putri" />

<meta name="date" content="2023-11-08" />

<title>sc_data_analysis_withCloneDetective</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NextClone-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/phipsonlab/NextClone-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">sc_data_analysis_withCloneDetective</h1>
<h4 class="author">Givanna Putri</h4>
<h4 class="date">2023-11-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-11-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NextClone-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231011code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231011)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231011code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231011)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabNextCloneanalysistreeca5457a58e1cb67a618a453b0da4ea502d30c771targetblankca5457aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/phipsonlab/NextClone-analysis/tree/ca5457a58e1cb67a618a453b0da4ea502d30c771" target="_blank">ca5457a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabNextCloneanalysistreeca5457a58e1cb67a618a453b0da4ea502d30c771targetblankca5457aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/phipsonlab/NextClone-analysis/tree/ca5457a58e1cb67a618a453b0da4ea502d30c771" target="_blank">ca5457a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/cellranger_out/
    Ignored:    data/nextclone_out/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/sc_data_analysis.Rmd</code>) and
HTML (<code>docs/sc_data_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/ca5457a58e1cb67a618a453b0da4ea502d30c771/analysis/sc_data_analysis.Rmd" target="_blank">ca5457a</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/NextClone-analysis/ca5457a58e1cb67a618a453b0da4ea502d30c771/docs/sc_data_analysis.html" target="_blank">ca5457a</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/44218c126b237dcc75bdb0d59db17741d35864cf/analysis/sc_data_analysis.Rmd" target="_blank">44218c1</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-10
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/NextClone-analysis/44218c126b237dcc75bdb0d59db17741d35864cf/docs/sc_data_analysis.html" target="_blank">44218c1</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-10
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/24b5a73d3e7662430104f5e58a4f1a9b188fc2ed/analysis/sc_data_analysis.Rmd" target="_blank">24b5a73</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-09
</td>
<td>
wflow_remove("analysis/sc_data_analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/NextClone-analysis/24b5a73d3e7662430104f5e58a4f1a9b188fc2ed/docs/sc_data_analysis.html" target="_blank">24b5a73</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-09
</td>
<td>
wflow_remove("analysis/sc_data_analysis.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/1662b0c96deb5f5949d8935c8a77990ddea7cb69/analysis/sc_data_analysis.Rmd" target="_blank">1662b0c</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-10-20
</td>
<td>
analysis for ozsinglecell 23
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/NextClone-analysis/1662b0c96deb5f5949d8935c8a77990ddea7cb69/docs/sc_data_analysis.html" target="_blank">1662b0c</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-10-20
</td>
<td>
analysis for ozsinglecell 23
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Analysis for scRNA-seq data of MCF7 cell line tagged with ClonMapper
protocol.</p>
<pre class="r"><code>library(data.table)
library(DropletUtils)
library(CloneDetective)
library(scater)</code></pre>
<p>Load data</p>
<pre class="r"><code>raw_clone_data &lt;- fread(&quot;data/nextclone_out/sc_clone_barcodes_20231109.csv&quot;)</code></pre>
<p>Convert to expression</p>
<pre class="r"><code>cell_by_clone_mat &lt;- generate_cell_clone_barcode_matrix(cell_clone_bcode_dt = raw_clone_data)</code></pre>
<p>Incorporate the cell by gene matrix.</p>
<pre class="r"><code>sce &lt;- read10xCounts(&quot;data/cellranger_out/filtered_feature_bc_matrix&quot;)</code></pre>
<pre><code>as(&lt;dgTMatrix&gt;, &quot;dgCMatrix&quot;) is deprecated since Matrix 1.5-0; do as(., &quot;CsparseMatrix&quot;) instead</code></pre>
<pre class="r"><code>sce</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 36601 7828 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames: NULL
colData names(2): Sample Barcode
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):</code></pre>
<p>Compute few simple metrics like average library size per cell.
Compute total transcript molecules detected per cell then compute
average based on the number of cells detected.</p>
<pre class="r"><code>cell_qc_metrics &lt;- perCellQCMetrics(sce)
summary(cell_qc_metrics$sum)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    639    2769   43798   50807   72781  613781 </code></pre>
<pre class="r"><code>summary(cell_qc_metrics$detected)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    185    1292    7142    5939    8790   13766 </code></pre>
<p>Get the 10x cell barcode.</p>
<pre class="r"><code>valid_cells_10x &lt;- colData(sce)$Barcode</code></pre>
<p>Tree map to show the proportion of 10x cells that have 0, 1, 2, ..
clones found.</p>
<pre class="r"><code>plt &lt;- draw_treemap(
    cell_by_clone_matrix = cell_by_clone_mat,
    valid_cells_bcodes = valid_cells_10x
)
plt</code></pre>
<p><img src="figure/sc_data_analysis.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/44218c126b237dcc75bdb0d59db17741d35864cf/docs/figure/sc_data_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">44218c1</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/24b5a73d3e7662430104f5e58a4f1a9b188fc2ed/docs/figure/sc_data_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">24b5a73</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-11-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/phipsonlab/NextClone-analysis/blob/1662b0c96deb5f5949d8935c8a77990ddea7cb69/docs/figure/sc_data_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">1662b0c</a>
</td>
<td>
Givanna Putri
</td>
<td>
2023-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="assign-clone-barcodes-to-cells" class="section level1">
<h1>Assign clone barcodes to cells</h1>
<pre class="r"><code>sce_with_clone &lt;- assign_and_embed_clones(
    cell_by_gene_mat = sce,
    cell_clone_reads_dt = raw_clone_data,
)
colData(sce_with_clone)</code></pre>
<pre><code>DataFrame with 7828 rows and 4 columns
                     Sample            Barcode        clone_barcode
                &lt;character&gt;        &lt;character&gt;          &lt;character&gt;
1    data/cellranger_out/.. AAACCCAGTAATGCTC-1                   NA
2    data/cellranger_out/.. AAACCCAGTATTTCCT-1                   NA
3    data/cellranger_out/.. AAACCCAGTTCGGACC-1 GTAATTGATGAGACTGCAAT
4    data/cellranger_out/.. AAACCCATCATCGCCT-1                   NA
5    data/cellranger_out/.. AAACCCATCGCCTATC-1 CGAGCTAAGTTTGTCCAGGT
...                     ...                ...                  ...
7824 data/cellranger_out/.. TTTGTTGAGACGCTCC-1 ACTTTGTCTAGATGTATAGA
7825 data/cellranger_out/.. TTTGTTGCACTCAAGT-1 TAGTCGGGTTGTTACGCGTT
7826 data/cellranger_out/.. TTTGTTGCATCGTCCT-1 TGGTTTCTATTGTCTAGTGC
7827 data/cellranger_out/.. TTTGTTGTCACTCACC-1                   NA
7828 data/cellranger_out/.. TTTGTTGTCTTCCTAA-1                   NA
          clone_barcode_criteria
                        &lt;factor&gt;
1    no_clones_found            
2    no_clones_found            
3    single_clone               
4    no_clones_found            
5    dominant_clone_moreThan_0_5
...                          ...
7824 dominant_clone_moreThan_0_5
7825 single_clone               
7826 single_clone               
7827 no_clones_found            
7828 no_clones_found            </code></pre>
<p>Count how many cells assigned to most dominant clones.</p>
<pre class="r"><code>clone_bcode_criteria &lt;- as.data.table(colData(sce_with_clone))
clone_bcode_criteria &lt;- data.table(table(clone_bcode_criteria$clone_barcode_criteria))

# proportion. of cells with multiple clone barcodes?
clone_bcode_criteria[, multiclone := ! V1 %in% c(&quot;single_clone&quot;, &quot;no_clones_found&quot;)]
clone_bcode_criteria[, .(prop = sum(N) / sum(clone_bcode_criteria$N)), by = multiclone]</code></pre>
<pre><code>   multiclone      prop
1:       TRUE 0.2599642
2:      FALSE 0.7400358</code></pre>
<pre class="r"><code># proportion of multiclones assigned to most dominant clone?
clone_bcode_criteria[, .(prop = N / sum(N)), by = multiclone]</code></pre>
<pre><code>   multiclone      prop
1:       TRUE 0.3395577
2:       TRUE 0.6604423
3:      FALSE 0.4859313
4:      FALSE 0.5140687</code></pre>
<p>Export the clone assignments as data.table which can later be
saved.</p>
<pre class="r"><code>clone_assignments &lt;- assign_and_embed_clones(
    cell_by_gene_mat = sce,
    cell_clone_reads_dt = raw_clone_data,
    embed_to_mat = FALSE
)
head(clone_assignments)</code></pre>
<pre><code>          CellBarcode         CloneBarcode     criteria
1: AAACCCAGTTCGGACC-1 GTAATTGATGAGACTGCAAT single_clone
2: AAACGAACATAGATGA-1 GTCATGTCAAGCAGTGGCGT single_clone
3: AAACGCTCAGCGACCT-1 AGACAGGGATGAGATATTCG single_clone
4: AAACGCTGTGTGCCTG-1 GGCCGCAGGTTATACATCAT single_clone
5: AAAGAACGTTGCGGAA-1 ACGTAGATGTAGAGTATGAA single_clone
6: AAAGGATAGAGCATAT-1 AAAGTCCGCTCCCGATAGTT single_clone</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.0

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scater_1.26.1               ggplot2_3.4.1              
 [3] scuttle_1.8.4               CloneDetective_0.1.0       
 [5] DropletUtils_1.18.1         SingleCellExperiment_1.20.0
 [7] SummarizedExperiment_1.28.0 Biobase_2.58.0             
 [9] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
[11] IRanges_2.32.0              S4Vectors_0.36.1           
[13] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[15] matrixStats_0.63.0          data.table_1.14.8          
[17] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] treemapify_2.5.6          bitops_1.0-7             
 [3] fs_1.6.1                  RColorBrewer_1.1-3       
 [5] httr_1.4.4                rprojroot_2.0.3          
 [7] tools_4.2.3               bslib_0.4.2              
 [9] utf8_1.2.3                R6_2.5.1                 
[11] irlba_2.3.5.1             vipor_0.4.5              
[13] HDF5Array_1.26.0          colorspace_2.1-0         
[15] rhdf5filters_1.10.0       withr_2.5.0              
[17] gridExtra_2.3             tidyselect_1.2.0         
[19] processx_3.8.0            compiler_4.2.3           
[21] git2r_0.31.0              cli_3.6.1                
[23] BiocNeighbors_1.16.0      DelayedArray_0.24.0      
[25] sass_0.4.5                scales_1.2.1             
[27] callr_3.7.3               stringr_1.5.0            
[29] digest_0.6.31             rmarkdown_2.20           
[31] R.utils_2.12.2            XVector_0.38.0           
[33] pkgconfig_2.0.3           htmltools_0.5.4          
[35] sparseMatrixStats_1.10.0  highr_0.10               
[37] fastmap_1.1.0             limma_3.54.1             
[39] rlang_1.0.6               rstudioapi_0.14          
[41] DelayedMatrixStats_1.20.0 farver_2.1.1             
[43] jquerylib_0.1.4           generics_0.1.3           
[45] jsonlite_1.8.4            BiocParallel_1.32.5      
[47] dplyr_1.1.0               R.oo_1.25.0              
[49] RCurl_1.98-1.10           magrittr_2.0.3           
[51] BiocSingular_1.14.0       GenomeInfoDbData_1.2.9   
[53] Matrix_1.5-3              Rcpp_1.0.10              
[55] ggbeeswarm_0.7.1          munsell_0.5.0            
[57] Rhdf5lib_1.20.0           fansi_1.0.4              
[59] ggfittext_0.10.1          viridis_0.6.2            
[61] lifecycle_1.0.3           R.methodsS3_1.8.2        
[63] stringi_1.7.12            whisker_0.4.1            
[65] yaml_2.3.7                edgeR_3.40.2             
[67] zlibbioc_1.44.0           rhdf5_2.42.0             
[69] grid_4.2.3                ggrepel_0.9.3            
[71] parallel_4.2.3            promises_1.2.0.1         
[73] dqrng_0.3.0               lattice_0.20-45          
[75] beachmat_2.14.0           locfit_1.5-9.7           
[77] knitr_1.42                ps_1.7.2                 
[79] pillar_1.8.1              codetools_0.2-19         
[81] ScaledMatrix_1.6.0        glue_1.6.2               
[83] evaluate_0.20             getPass_0.2-2            
[85] vctrs_0.5.2               httpuv_1.6.9             
[87] purrr_1.0.1               gtable_0.3.1             
[89] cachem_1.0.6              xfun_0.39                
[91] rsvd_1.0.5                later_1.3.0              
[93] viridisLite_0.4.1         tibble_3.1.8             
[95] beeswarm_0.4.0           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
