---
title: "NextClone analysis"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Abstract {-}

The study of clonal dynamics has significantly advanced our understanding of cellular heterogeneity and lineage trajectories. With recent developments in lineage-tracing protocols such as ClonMapper or SPLINTR, which combine DNA barcoding with single-cell RNA sequencing (scRNA-seq), biologists can trace the lineage and evolutionary paths of individual clones while simultaneously observing their transcriptomic changes over time. Here, we present NextClone and CloneDetective, an integrated highly scalable Nextflow pipeline and R package for efficient extraction and quantification of clonal barcodes from scRNA-seq data and DNA sequencing data tagged with lineage-tracing barcodes. We applied both NextClone and CloneDetective to data from a barcoded MCF7 cell line, and demonstrate their important role for advancing clonal analysis in the era of high-throughput sequencing.

NextClone and CloneDetective are freely available and open-source on github (https://github.com/phipsonlab/NextClone and https://github.com/phipsonlab/CloneDetective). Documentations and tutorials for both NextClone and CloneDetective can be found at https://phipsonlab.github.io/NextClone/index.html and https://phipsonlab.github.io/CloneDetective/index.html respectively.

# Authors {-}

Givanna Putri ^1^, Nichelle Pires ^2^, Nadia Davidson ^1^, Catherine Blyth ^2^, Aziz Alâ€™Khafaji ^3^, Shom Goel ^2^, Belinda Phipson ^1^

^1^ The Walter and Eliza Hall Institute of Medical Research, Parkville, VIC, Australia

^2^ Peter MacCallum Cancer Centre and The Sir Peter MacCallum Department of Oncology, The University of Melbourne, Parkville, VIC, Australia

^3^ Broad Institute of MIT and Harvard, Cambridge, Massachusetts, USA

## Analysis

The FASTQ files for the DNA-seq data are available on zenodo: https://zenodo.org/doi/10.5281/zenodo.10121942.

The BAM file for the scRNA-seq data is available on zenodo: https://zenodo.org/doi/10.5281/zenodo.10129133 and https://zenodo.org/doi/10.5281/zenodo.10129624

The BAM file for the scRNA-seq data was obtained by running cellranger version 7.1.0 
aligning to hg38 genome, specifically the GRCh38-2020-A available on https://www.10xgenomics.com/support/software/cell-ranger/downloads.

1. [DNA-seq data](DNAseq_data_analysis.html)
2. [scRNA-seq data](sc_data_analysis.html)
3. [runtime benchmarking](benchmarking_duration.html)


