---
title: "scRNA-seq data analysis"
author: "Givanna Putri"
date: "2023-11-08"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Analysis for scRNA-seq data of MCF7 cell line tagged with ClonMapper protocol.

```{r message=FALSE, warning=FALSE}
library(data.table)
library(DropletUtils)
library(CloneDetective)
library(scater)
```

## Load data

The command used to generate NextClone output is available in code folder.

```{r}
raw_clone_data <- fread("data/nextclone_out/sc_clone_barcodes_20231117.csv")
```


## Generate cell by clone matrix

```{r}
cell_by_clone_mat <- generate_cell_clone_barcode_matrix(cell_clone_bcode_dt = raw_clone_data)
```

## Load the cell by gene matrix

This matrix is the filtered count matrix generated using cellranger version 7.1.0
aligning to hg38 genome, specifically the GRCh38-2020-A available on https://www.10xgenomics.com/support/software/cell-ranger/downloads.

The following Nextflow pipeline was used:

```{r eval=FALSE}
#!/usr/bin/env nextflow

process run_cellranger_count {
    cpus 24
    memory '100 GB'
    time '24 hours'
    module 'cellranger/7.1.0'
    publishDir "$params.publish_dir",  mode: 'copy'

    input:
        val sample_name
    
    output:
        path "${sample_name}"

    """
    cellranger count --id=${sample_name} \
                     --sample=${sample_name} \
                     --transcriptome=${params.ref_fasta} \
                     --fastqs=${params.fastqs_dir} \
                     --localcores=${task.cpus} \
                     --localmem=${task.memory.toGiga()} \
                     --nosecondary
    """
}

workflow {
    run_cellranger_count("PilotDataset")
}
```


```{r}
sce <- read10xCounts("data/cellranger_out/filtered_feature_bc_matrix")
sce
```

Compute few simple metrics like average library size per cell.
Compute total transcript molecules detected per cell then compute average
based on the number of cells detected.

```{r}
cell_qc_metrics <- perCellQCMetrics(sce)
summary(cell_qc_metrics$sum)
summary(cell_qc_metrics$detected)
```


Get the 10x cell barcode.

```{r}
valid_cells_10x <- colData(sce)$Barcode
```

Tree map to show the proportion of 10x cells that have 0, 1, 2, .. clones found.

```{r}
plt <- draw_treemap(
    cell_by_clone_matrix = cell_by_clone_mat,
    valid_cells_bcodes = valid_cells_10x
)
plt
```

## Assign clone barcodes to cells

```{r}
sce_with_clone <- assign_and_embed_clones(
    cell_by_gene_mat = sce,
    cell_clone_reads_dt = raw_clone_data,
)
colData(sce_with_clone)

```

Interrogate the clone assignments further.

```{r}
clone_bcode_criteria <- as.data.table(colData(sce_with_clone))
clone_bcode_criteria <- data.table(table(clone_bcode_criteria$clone_barcode_criteria))
setnames(clone_bcode_criteria, "V1", "criteria")
```

What proportion of cells where multiple clone barcodes were detected?

```{r}
clone_bcode_criteria[, prop := N / dim(sce)[2]]
clone_bcode_criteria
```

What proportion of multiclone cells assigned the most dominant clone barcode?

```{r}
n_multiclone <- clone_bcode_criteria[criteria %in% c("clone_from_edit_distance", "dominant_clone_moreThan_0_5")]
n_multiclone[, prop := N / sum(n_multiclone$N)]
n_multiclone
```

Export the clone assignments as data.table which can later be saved.

```{r}
clone_assignments <- assign_and_embed_clones(
    cell_by_gene_mat = sce,
    cell_clone_reads_dt = raw_clone_data,
    embed_to_mat = FALSE
)
head(clone_assignments)
```

